msteams-platform/messaging-extensions/api-based-overview.md

```diff
@@ -8,79 +8,574 @@ ms.author: anclear
 
 * Retrieve real-time information, such as latest news coverage on a product launch.
-* Retrieve knowledge-based information, for example, my team’s design files in Figma.
-
-You can create an API-based message extension using an [OpenAPI Description (OAD)](https://learn.openapis.org/specification/) document. After you've created an OpenAPI Description document, use the OpenAPI Description document to generate and integrate the client code in your app's project. Create or generate a response rendering template to manage the responses from the API.
+* Retrieve knowledge-based information, for example, my team’s design files in Figma
 
 See the video to learn more about building an API-based message extension using Teams Toolkit:
 </br>
 </br>
 
 > [!VIDEO https://www.youtube.com/embed/jSYNHz6hz4Y?si=htmfWtlY9bYH_RT2]
 
+<br>
+
+
+|Traditional bot-based message extensions  |API-based message extensions  |
+|---------|---------|
+|Developers need to build, deploy, and maintain a service to handle invoke commands from the Teams client.     | If the end-service's APIs can be described using the OpenAPI Specification, developers can eliminate the need for the middle-layer handling service.         |
+|This service processes the incoming query and makes a call to the developer’s end-service.     | Teams can directly use the [OpenAPI Specification](https://swagger.io/resources/open-api/) to build requests and communicate with the developer's end-service.        |
+
+The following images show the flow of user queries through Traditional message extensions and API message extensions:
+
+:::image type="content" source="../assets/images/Copilot/api-based-me-flow.png" alt-text="Screenshot shows user query flow between a user, Teams Client, and Teams bot service using Traditional message extensions. The diagram also shows how the API spec, the rendering templates, the API relate to each other." lightbox="../assets/images/Copilot/api-based-me-flow.png":::
+*User query flow using Traditional Message Extensions. The developer must maintain a custom bot handler service, which handles the requests from a Teams bot. The handler service sends a request to the developer’s service when a query is invoked.*
+
+<br>
+
+:::image type="content" source="../assets/images/Copilot/api-based-me-flow-2.png" alt-text="Screenshot shows the query flow between a user, Teams Client, and Teams bot service using API Message Extensions. The diagram also shows how the API spec, the rendering templates, the API relate to each other." lightbox="../assets/images/Copilot/api-based-me-flow-2.png":::
+*User query flow using API Message Extensions. There's no need for a developer maintained handler service as long as the interaction is clearly outlined in the OpenAPI Specification in the App Package.*
+
+
+<br>
```

Comment:
I think we missed a paragraph:
![image](https://github.com/user-attachments/assets/5be2a81e-675a-4da6-a1db-91e4e9deaa53)

Reply:
nvm looks like it comes later on.